### ISSUE TITLE
Bug: Cannot add labels to default PVCs created by the StatefulSet in MongoDB Community CR

### ISSUE THREAD

**Issue by Ahmad-Faizan** [mongodb/mongodb-kubernetes-operator#1165]
**What did you do to encounter the bug?**
1. Install the [MongoDB Community Operator Helm chart](https://github.com/mongodb/helm-charts/blob/main/charts/community-operator)
2. Create a `kind: MongoDB` resource with stateful set

```yaml
apiVersion: mongodbcommunity.mongodb.com/v1
kind: MongoDBCommunity
metadata:
  name: mongodb-test
  namespace: mongodb-operator
spec:
  members: 2
  type: ReplicaSet
  version: 4.4.0
  statefulSet:
    spec:
      volumeClaimTemplates:
      - metadata:
          name: data-volume
          labels:
            velero.io/exclude-from-backup: "true"
        spec:
          accessModes: ["ReadWriteOnce"]
          resources:
            requests:
              storage: 1Gi
      - metadata:
          name: logs-volume
          labels:
            velero.io/exclude-from-backup: "true"
        spec:
          accessModes: ["ReadWriteOnce"]
          resources:
            requests:
              storage: 500Mi
  security:
    tls:
      enabled: false
    authentication:
      modes: ["SCRAM"]
  users: []
```

4. Apply the above YAML using `kubectl apply -f <path/to/file>`
5. List the PVCs created in the `mongodb-operator` namespace
6. Check the labels on the created PVCs - `data-volume` and `logs-volume`. The label `velero.io/exclude-from-backup: "true"` is not present in the PVC.

**What did you expect?**
MongoDB should have created the `data-volume` and `logs-volume` with the velero label.

**What happened instead?**
MongoDB created the PVC from the stateful set spec, but did not add the labels.

**Screenshots**

**Operator Information**
 - Operator Version: 0.7.6
 - MongoDB Image used: MongoDB 4.4.0

**Kubernetes Cluster Information**
 - Distribution
 - Version: 1.22.3
 - Image Registry location (quay, or an internal registry): quay.io

**Additional context**
The community operator Helm chart creates PVCs but there is no way to add labels on the PVCs being created.
The reason we want to do this is to exclude the PVCs from being backed up by Velero. Velero expects a label on the resources which don't need to be backed up. To avoid the MongoDB PVC in the backups, we need to add a label to the PVCs.

I found these lines in the CRD definition which relate to the stateful set
https://github.com/mongodb/helm-charts/blob/main/charts/community-operator-crds/templates/mongodbcommunity.mongodb.com_mongodbcommunity.yaml#L319-L328

If possible, please include:
 - `kubectl describe` output
 - yaml definitions for your objects
 - log files for the operator and database pods

**Comment by dan-mckean** [comment]
Hi, I'm Dan and I'm the Product Manager for MongoDB's support of Kubernetes.

I'm doing some work right now to try and identify how the Community Operator is being used. The Community Operator is something I inherited when I started at MongoDB, but it doesn't get as much attention from us as we'd like and we're trying to understand how it's used in order to establish its future. It will help us prioritize future issues and PRs raised by the community 🙂

Here's a super short survey (it's much easier for us to review all the feedback that way!): [https://docs.google.com/forms/d/e/1FAIpQLSfwrwyxBSlUyJ6AmC-eYlgW_3JEdfA48SB2i5--_WpiynMW2w/viewform?usp=sf_link](https://docs.google.com/forms/d/e/1FAIpQLSfwrwyxBSlUyJ6AmC-eYlgW_3JEdfA48SB2i5--_WpiynMW2w/viewform?usp=sf_link)

 If you'd rather email me instead: [dan.mckean@mongodb.com](mailto:dan.mckean@mongodb.com?subject=MongoDB%20Community%20Operator%20feedback)

Thank you in advance!
Dan
